src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -166,32 +175,179 @@ public void CompilerDataLogger_Dispose_ShouldReadSarifV1()
 
             fileSystem.Verify(fileSystem => fileSystem.FileOpenRead(sarifLogPath), Times.Once);
             sendItems.Count.Should().Be(1);
+            context.Dispose();
         }
 
-        private List<ITelemetry> TestSetup(string sarifLogFilePath, BinaryAnalyzerContext context, Sarif.SarifVersion sarifVersion, out CompilerDataLogger logger, IFileSystem fileSystem = null)
+        [Fact]
+        public void CompilerDataLogger_Summarize_ShouldLogSummaryEvent()
         {
-            List<ITelemetry> sendItems = null;
-            TelemetryClient telemetryClient;
-            TelemetryConfiguration telemetryConfiguration;
+            string sarifLogPath = Path.Combine(PEBinaryTests.BaselineTestDataDirectory, ExpectedFolder, SampleSarifPath);
+            var fileSystem = new Mock<IFileSystem>();
+            BinaryAnalyzerContext context = CreateTestContext();
+            List<ITelemetry> sendItems = TestSetup(sarifLogPath, context, Sarif.SarifVersion.Current, out CompilerDataLogger compilerDataLogger, fileSystem.Object);
+            SarifLog sarifLog = SarifLog.Load(sarifLogPath);
+            ExecutionException exception = AnalysisSummaryExtractor.ExtractExceptionData(sarifLog).FirstOrDefault();
+            AnalysisSummary summary = AnalysisSummaryExtractor.ExtractAnalysisSummary(sarifLog, "", null);
 
-            sendItems = new List<ITelemetry>();
+            compilerDataLogger.Summarize(summary);
 
-            telemetryConfiguration = new TelemetryConfiguration();
-            telemetryConfiguration.InstrumentationKey = Guid.NewGuid().ToString();
-            telemetryConfiguration.TelemetryChannel = new StubTelemetryChannel { OnSend = item => sendItems.Add(item) };
-            telemetryConfiguration.TelemetryInitializers.Add(new OperationCorrelationTelemetryInitializer());
+            fileSystem.Verify(fileSystem => fileSystem.FileOpenRead(sarifLogPath), Times.Never);
+            sendItems.Count.Should().Be(1);
```

Comment:
> sendItems.Count.Should().Be(1); [](http://example.com/codeflow?start=11&length=32)

Add verification in these more utility methods such as summarize / dispose, etc that manipulate both disk and telemetry.

Reply:
Not adding additional disk validations at this time.